modules/calib3d/include/opencv2/calib3d.hpp

```diff
@@ -1080,23 +1101,38 @@ The algorithm performs the following steps:
    findChessboardCorners, solvePnP, initCameraMatrix2D, stereoCalibrate, undistort
  */
 CV_EXPORTS_AS(calibrateCameraExtended) double calibrateCamera( InputArrayOfArrays objectPoints,
-                                     InputArrayOfArrays imagePoints, Size imageSize,
+                                     InputArrayOfArrays imagePoints, Size imageSize, int iFixedPoint,
```

Comment:
Lets preserve existing interfaces and add new one (something like `calibrateCameraReleaseObject`).

> iFixedPoint

Looks like, this and other new parameters imply `flags += CALIB_RELEASE_OBJECT` automatically.

Reply:
> Lets preserve existing interfaces and add new one (something like `calibrateCameraReleaseObject`).

That's my initial thought before starting. With regard to the fact that
they share most of the interface documents, I chose functions overload.
OK, I'll add new interfaces instead of overload old ones. How about the
name `calibrateCameraRO`? It's shorter and attractive to users, but with
the meaning less obvious. Since most part of the interface document are
shared by `calibrateCamera` and `calibrateCameraRO`,
`calibrateCameraRO`'s interface document will be simpler with most
explanations referenced to `calibrateCamera`.

The standard calibration method and object-releasing calibration method
will be implemented in `calibrateCameraRO`, and method is chosen based
on whether `iFixedPoint` is within a rational range. `calibrateCamera`
is then made a wrapper to `calibrateCameraRO`.

> 
> > iFixedPoint
> 
> Looks like, this and other new parameters imply `flags += CALIB_RELEASE_OBJECT` automatically.

Only `iFixedPoint` is input parameter, the other two are optional output
parameters. You are right. The method selection can be controlled with
`iFixedPoint`. I'll remove `CALIB_RELEASE_OBJECT` item and make the
method selection based on the value of `iFixedPoint`.